src/type_description.ml

```diff
@@ -0,0 +1,1781 @@
+    let vulkan = F.constant "SDL_WINDOW_VULKAN" F.uint32_t
+
+    let pos_centered = F.constant "SDL_WINDOWPOS_CENTERED" F.int
+    let pos_undefined = F.constant "SDL_WINDOWPOS_UNDEFINED" F.int
+  end
+
+  module Gl = struct
+
+    (* contextFlag *)
+
+    let context_debug_flag = F.constant "SDL_GL_CONTEXT_DEBUG_FLAG" F.int
+    let context_forward_compatible_flag = F.constant "SDL_GL_CONTEXT_FORWARD_COMPATIBLE_FLAG" F.int
+    let context_robust_access_flag = F.constant "SDL_GL_CONTEXT_ROBUST_ACCESS_FLAG" F.int
+    let context_reset_isolation_flag = F.constant "SDL_GL_CONTEXT_RESET_ISOLATION_FLAG" F.int
+    let context_release_behavior = F.constant "SDL_GL_CONTEXT_RELEASE_BEHAVIOR" F.int
+
+    (* profile *)
+
+    let context_profile_core = F.constant "SDL_GL_CONTEXT_PROFILE_CORE" F.int
+    let context_profile_compatibility = F.constant "SDL_GL_CONTEXT_PROFILE_COMPATIBILITY" F.int
+    let context_profile_es = F.constant "SDL_GL_CONTEXT_PROFILE_ES" F.int
+
+    (* attr *)
+
+    let red_size = F.constant "SDL_GL_RED_SIZE" F.int
+    let green_size = F.constant "SDL_GL_GREEN_SIZE" F.int
+    let blue_size = F.constant "SDL_GL_BLUE_SIZE" F.int
+    let alpha_size = F.constant "SDL_GL_ALPHA_SIZE" F.int
+    let buffer_size = F.constant "SDL_GL_BUFFER_SIZE" F.int
+    let doublebuffer = F.constant "SDL_GL_DOUBLEBUFFER" F.int
+    let depth_size = F.constant "SDL_GL_DEPTH_SIZE" F.int
+    let stencil_size = F.constant "SDL_GL_STENCIL_SIZE" F.int
+    let accum_red_size = F.constant "SDL_GL_ACCUM_RED_SIZE" F.int
+    let accum_green_size = F.constant "SDL_GL_ACCUM_GREEN_SIZE" F.int
+    let accum_blue_size = F.constant "SDL_GL_ACCUM_BLUE_SIZE" F.int
+    let accum_alpha_size = F.constant "SDL_GL_ACCUM_ALPHA_SIZE" F.int
+    let stereo = F.constant "SDL_GL_STEREO" F.int
+    let multisamplebuffers = F.constant "SDL_GL_MULTISAMPLEBUFFERS" F.int
+    let multisamplesamples = F.constant "SDL_GL_MULTISAMPLESAMPLES" F.int
+    let accelerated_visual = F.constant "SDL_GL_ACCELERATED_VISUAL" F.int
+    let context_major_version = F.constant "SDL_GL_CONTEXT_MAJOR_VERSION" F.int
+    let context_minor_version = F.constant "SDL_GL_CONTEXT_MINOR_VERSION" F.int
+    let context_egl = F.constant "SDL_GL_CONTEXT_EGL" F.int
+    let context_flags = F.constant "SDL_GL_CONTEXT_FLAGS" F.int
+    let context_profile_mask = F.constant "SDL_GL_CONTEXT_PROFILE_MASK" F.int
+    let share_with_current_context = F.constant "SDL_GL_SHARE_WITH_CURRENT_CONTEXT" F.int
+    let framebuffer_srgb_capable = F.constant "SDL_GL_FRAMEBUFFER_SRGB_CAPABLE" F.int
+
+    type _context
+    type context = _context Ctypes_static.structure
+    let context : context F.typ = F.structure "SDL_GLContext"
+  end
+
+  module Vulkan = struct
+    type _surface
+    type surface = _surface Ctypes_static.structure Ctypes_static.ptr
+
+    let raw_surface : _surface Ctypes_static.structure F.typ =
+      F.structure "VkSurfaceKHR_T"
+    let surface : surface F.typ = F.ptr raw_surface
+  end
```

Comment:
Require `<SDL_Vulkan.h>` which makes me wonder if it is an independent .h because it's not always available and therefore should be optionally bound (how?).
did not complain that it was missing under mingw windows though... Maybe worth trying under MacOS/with msvc...